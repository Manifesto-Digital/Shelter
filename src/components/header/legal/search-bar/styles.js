import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../../theme/breakpoint';
import { buttonReset } from '../../../styled/buttons';
import { inputStyles } from '../../../styled/inputs';

export const FormWrapper = styled.form`
  display: none;
  height: 43px;

  ${breakpoint.desktop`
      display: flex;
  `}
`;

export const InputWrap = styled.div`
  display: flex;
  position: relative;
  margin-right: 5px;
  font-size: ${({ theme }) => theme.fontsize.small};
  width: 100%;
  margin: 0 ${({ theme }) => theme.spacing.small};
  border: 1px solid ${({ theme }) => theme.palette.grey10};

  ${breakpoint.desktop`
    width: auto;
    border: 0;
  `}
`;

export const SearchInput = styled.input`
  ${inputStyles}
  width: 100%;
  margin-right: 5px;
  background-color: ${({ theme }) => theme.palette.white};
  border: 1px solid ${({ theme }) => theme.palette.black};
  padding-left: ${({ theme }) => theme.spacing.large};

  ${breakpoint.desktop`
      min-width: 320px;
  `}
`;

export const SearchIcon = styled(SVG)`
  display: block;
  position: absolute;
  left: 15px;
  width: 18px;
  top: 50%;
  transform: translateY(-50%);
  fill: ${({ theme }) => theme.palette.grey66};

  svg {
    display: block;
  }
`;

export const SearchButton = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 18px;
  color: ${({ theme }) => theme.palette.white};
  background-color: ${({ theme }) => theme.palette.grey80};
`;
