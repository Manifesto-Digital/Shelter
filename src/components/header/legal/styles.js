import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';
import { buttonStyles, buttonReset } from '../../styled/buttons';
import LinkHandler from '../../link-handler';

export const LegalDonateButton = styled(LinkHandler)`
  ${buttonStyles};
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: ${({ mobileMenu, theme }) => mobileMenu && theme.spacing.small};
  text-decoration: none;

  ${breakpoint.desktop`
    height: 100%;
  `};

  &:hover {
    color: ${({ theme }) => theme.palette.white};
  }
`;

export const LegalMenuList = styled.ul`
  display: flex;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
`;

export const SecondHeaderWrapper = styled.div`
  display: none;
  padding: ${({ theme }) => theme.spacing.small} 0;
  background-color: ${({ theme }) => theme.palette.grey25};
  ${breakpoint.desktop`
    display: block;
  `};
`;

export const SecondHeaderBar = styled.div`
  display: flex;
  align-items: center;
  margin-bottom: ${({ theme }) => theme.spacing.small};
`;

export const SubsectionHeader = styled.h3`
  margin-right: ${({ theme }) => theme.spacing.large};
  margin-bottom: 0;
  font-weight: normal;
`;

export const LegalItem = styled.li`
  display: flex;
  align-items: center;
  position: relative;
  margin: 0 ${({ theme }) => theme.spacing.standard} 0 0;
  font-size: ${({ theme }) => theme.fontsize.small};
`;

export const MoreItemMenu = styled.button`
  ${buttonReset};
  font-size: ${({ theme }) => theme.fontsize.small};
`;

export const ThirdNavigationBar = styled.div`
  display: flex;
  background-color: ${({ theme }) => theme.palette.grey25};
  padding: ${({ theme }) => theme.spacing.small} 0;

  a {
    font-weight: normal;
  }
`;

/* Currently only display on desktop */
export const MoreSubMenu = styled.ul`
  position: absolute;
  display: flex;
  flex-direction: column;
  top: calc(100% + ${({ theme }) => theme.spacing.standard});
  right: 0;
  margin: 0;
  min-width: 225px;
  padding: ${({ theme }) => theme.spacing.standard};
  border: none;
  list-style: none;
  background: ${({ theme }) => theme.palette.grey15};

  ${LegalItem} {
    margin-bottom: ${({ theme }) => theme.spacing.standard};
    &:last-child {
      border: none;
      margin-bottom: 0;
    }
  }
`;
