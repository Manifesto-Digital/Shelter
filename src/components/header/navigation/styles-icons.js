import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { buttonReset } from '../../styled/buttons';
import { breakpoint } from '../../theme/breakpoint';

export const MobileMenuClose = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  align-items: center;
  width: 45px;
  height: 40px;
  padding: 0;
  margin-left: auto;
  cursor: pointer;

  ${breakpoint.desktop`
    display: none;
  `}
`;

export const SubNavButton = styled(MobileMenuClose)`
  height: auto;
  width: 25px;

  ${breakpoint.desktop`
    display: ${({ legal }) => (legal ? 'none' : 'flex')};
    margin-left: 5px;

    svg {
      width: auto;
    }
  `}

  svg {
    width: 15px;
  }
`;

export const MenuSVG = styled(SVG)`
  display: block;
  width: 20px;
`;

export const ArrowSVG = styled(SVG)`
  display: block;
  fill: ${({ theme }) => theme.palette.black};
  width: 20px;
  cursor: pointer;
  pointer-events: none;
`;
