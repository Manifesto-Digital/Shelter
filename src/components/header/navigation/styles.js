import styled from 'styled-components';
import { breakpoint } from '../../theme/breakpoint';
import LinkHandler from '../../link-handler';

export const Wrapper = styled.div`
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  width: 80%;
  height: 100%;
  background: ${({ theme }) => theme.palette.white};
  box-shadow: ${({ theme }) => theme.boxshadow.menu};
  transition: transform 0.3s cubic-bezier(0, 0, 0.3, 1);
  transform: ${({ active }) =>
    active ? 'translateX(0)' : ' translateX(120%)'};
  z-index: ${({ active }) => (active ? '10' : 1)};
  overflow-y: scroll;

  ${breakpoint.desktop`
    position: relative;
    width:100%;
    background: ${({ theme }) => theme.palette.grey10};
    box-shadow: none;
    transition: none;
    transform: translateX(0);
    overflow-y: visible;
  `}
`;

export const MenuList = styled.ul`
  display: flex;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;

  ${breakpoint.desktop`
    flex-direction: row;
    width: auto;
  `}
`;

export const Menus = styled.nav`
  position: relative;
  display: flex;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 0;

  ${breakpoint.desktop`
    flex-direction: row;
  justify-content: ${({ legal }) => legal && 'flex-end'};
    max-width: ${({ theme }) => theme.sizes.maxWidth};
    margin: 0 auto;
    padding: 0 ${({ theme }) => theme.spacing.small};
  `}
`;

export const Item = styled.li`
  display: flex;
  align-items: center;
  margin-bottom: 0;
  padding: ${({ topLevel, legal, theme }) =>
    (legal && topLevel && `15px ${theme.spacing.standard}`) ||
    (topLevel && '0') ||
    `${theme.spacing.small}}`};
  border-bottom: ${({ topLevel, theme }) =>
    topLevel ? ' none' : `1px solid ${theme.palette.grey10}`};
  border-bottom: ${({ topLevel, legal, theme }) =>
    (legal && topLevel && `1px solid ${theme.palette.grey10}`) ||
    (topLevel && 'none') ||
    `1px solid ${theme.palette.grey10}`};

  ${MenuList} {
    padding: 5px ${({ theme }) => theme.spacing.small};
    ${breakpoint.desktop`
      padding:5px 0;
    `}

    ${breakpoint.desktop`
    margin-right: auto;
     padding: ${({ topLevel, theme }) =>
       topLevel ? `0` : `${theme.spacing.small}`};
  `}
  }
`;

export const ItemLink = styled(LinkHandler)`
  color: ${({ theme }) => theme.palette.black};
  font-size: ${({ theme }) => theme.fontsize.small};
  text-decoration: none;
`;

export const AdditionalMenu = styled(MenuList)`
  padding: 0 ${({ theme }) => theme.spacing.small};
  ${Item} {
    padding: 15px ${({ theme }) => theme.spacing.small};
    &:last-child {
      border: none;
      ${breakpoint.desktop`
        margin-left: ${({ theme }) => theme.spacing.standard};
      `}
    }
  }
  ${breakpoint.desktop`
    flex-direction: row;
    min-width: 200px;
    margin-left: auto;
    margin-right: 0;
      ${ItemLink} {
        border:none;
      }
  `}
`;

export const SubMenu = styled.ul`
  display: ${({ active }) => (active ? 'flex' : 'none')};
  flex-direction: column;
  padding: 0 ${({ theme }) => theme.spacing.small};
  border: none;
  list-style: none;

  ${breakpoint.desktop`
    position: absolute;
    display: ${({ legal }) => legal && 'none'};
    top: 52px;
    right: 0;
    left: 0;
    flex-direction: row;
    flex-wrap: wrap;
    width: 100%;
    height: auto;
    margin: 0;
    background: ${({ theme }) => theme.palette.grey15};
    padding: ${({ theme }) => theme.spacing.small} ${({ theme }) =>
    theme.spacing.small};
  `}

  ${Item} {
    &:last-child {
      border: none;
    }
    ${breakpoint.desktop`
      flex: 0 1 calc(${100 / 3}% - 10px);
      width: 32%;
      margin-left: ${({ theme }) => theme.spacing.small};
      border: none;
    `}

    &:hover {
      ${breakpoint.desktop`
          background: ${({ theme }) => theme.palette.black};
        ${ItemLink} {
          color: ${({ theme }) => theme.palette.white};
        }
      `}
    }
  }
`;

export const SkipToContent = styled.a`
  display: block;
  width: 100%;
  max-width: ${({ theme }) => theme.sizes.maxWidth};
  height: 0;
  margin: 0 auto;
  overflow: hidden;

  &:focus {
    height: auto;
  }
`;
