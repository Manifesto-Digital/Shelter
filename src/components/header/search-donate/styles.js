import styled from 'styled-components';
import SVG from 'react-inlinesvg';
import { breakpoint } from '../../theme/breakpoint';
import { buttonReset, buttonStyles } from '../../styled/buttons';
import { inputStyles } from '../../styled/inputs';
import LinkHandler from '../../link-handler';

import theme from '../../theme/variables';

export const InputWrap = styled.div`
  display: flex;
  margin-right: ${theme.spacing.xs};
  font-size: ${theme.fontsize.small};
  width: 100%;
  margin: 0 ${theme.spacing.small};
  border: 1px solid ${theme.palette.grey10};

  ${breakpoint.desktop`
    width: auto;
    border: 0;
  `}
`;

export const SearchInput = styled.input`
  ${inputStyles}
  border: 0;
  width: 100%;

  ${breakpoint.desktop`
      min-width: 185px;
  `}
`;

export const SearchButton = styled.button`
  ${buttonReset}
  display: flex;
  justify-content: center;
  align-items: center;
  padding: 0 ${theme.spacing.small};
  background-color: ${theme.palette.white};
`;

export const SearchIcon = styled(SVG)`
  display: block;
  width: 18px;
  fill: ${theme.palette.royalBlue};
`;

export const DonateButton = styled(LinkHandler)`
  ${buttonStyles};
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: ${({ mobileOnly }) => mobileOnly && theme.spacing.small};
  text-decoration: none;
  ${breakpoint.desktop`
    display: ${({ mobileOnly }) => (mobileOnly ? 'none' : 'flex')};
  `}

  &:hover {
    color: ${theme.palette.white};
  }
`;

export const Wrapper = styled.form`
  display: ${({ resolution }) => (resolution === 'desktop' ? 'none' : 'flex')};

  ${breakpoint.desktop`
      display: ${({ resolution }) =>
        resolution === 'desktop' ? 'flex' : 'none'};
  `}

  ${DonateButton} {
    display: ${({ resolution }) => (resolution === 'mobile' ? 'none' : 'flex')};
  }

  height: 43px;
`;
