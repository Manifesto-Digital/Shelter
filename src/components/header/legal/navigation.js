import React, { useState } from 'react';
import PropTypes from 'prop-types';
import NavigationMenuItem from '../navigation/menu-item';
import { Overlay } from '../../styled/overlay';
import { VisuallyHidden } from '../../styled/accessibility';
import { MobileMenuClose, MenuSVG } from '../navigation/styles-icons';
import {
  Wrapper,
  Menus,
  MenuList,
  Item,
  ItemLink,
  SkipToContent,
} from '../navigation/styles';
import { LegalDonateButton } from './styles';
import CloseSVG from '../../../assets/svg/icons/times-light.svg';
import LegalSearchBar from './search-bar';

const Navigation = ({
  navigationItems,
  active,
  searchFocus,
  openState,
  searchState,
}) => {
  const [activeMenu, setActiveMenu] = useState('');

  const updateActiveMenu = id => {
    if (id === activeMenu) {
      setActiveMenu(false);
    } else {
      setActiveMenu(id);
    }
  };

  return (
    <>
      <Wrapper active={active} ariaHidden={active}>
        <MobileMenuClose
          type="button"
          onClick={() => {
            openState();
            if (searchFocus) searchState();
          }}
          active={active}
          aria-expanded={active}
        >
          <MenuSVG src={CloseSVG} />
        </MobileMenuClose>
        <LegalSearchBar searchFocus={searchFocus} resolution="mobile" />
        <Menus role="navigation" aria-label="Main menu" legal>
          {navigationItems && (
            <MenuList role="menubar" aria-hidden="false">
              {navigationItems.map((item, i) => (
                <NavigationMenuItem
                  key={i}
                  id={item.id}
                  menuItem={item}
                  menuOpen={activeMenu === item.id}
                  setActiveMenu={() => updateActiveMenu(item.id)}
                  legal
                />
              ))}
              <Item topLevel desktopOnly>
                <LegalDonateButton
                  internalLink={{ slug: 'donate' }}
                  bg="donate"
                  noMargin
                >
                  Donate
                  <VisuallyHidden as="legend">Donate</VisuallyHidden>
                </LegalDonateButton>
              </Item>
              <Item topLevel lastItem>
                <ItemLink
                  as="a"
                  href="https://scotland.shelter.org.uk/"
                  rel="noopener noreferrer"
                  target="_blank"
                >
                  Scotland
                </ItemLink>
              </Item>
            </MenuList>
          )}
        </Menus>
        <SkipToContent href="#main">Skip to main content</SkipToContent>
      </Wrapper>
      <Overlay
        active={active}
        subMenuActive={!!activeMenu}
        onClick={() => updateActiveMenu()}
      />
    </>
  );
};

Navigation.propTypes = {
  active: PropTypes.bool.isRequired,
  searchFocus: PropTypes.bool,
  openState: PropTypes.func,
  searchState: PropTypes.func,
  navigationItems: PropTypes.array,
};

export default Navigation;
