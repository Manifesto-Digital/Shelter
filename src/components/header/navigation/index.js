import React, { useState } from 'react';
import PropTypes from 'prop-types';
import NavigationMenu from './navigation';
import SearchDonate from '../search-donate';
import { Overlay } from '../../styled/overlay';

import { MobileMenuClose, MenuSVG } from './styles-icons';
import {
  Wrapper,
  Menus,
  MenuList,
  AdditionalMenu,
  Item,
  ItemLink,
} from './styles';
import CloseSVG from '../../../assets/svg/icons/times-light.svg';

const Navigation = ({
  pageData,
  active,
  searchFocus,
  openState,
  searchState,
}) => {
  const [activeMenu, setActiveMenu] = useState('');
  const { navigationItems, additionalLink } = pageData;

  const updateActiveMenu = id => {
    if (id === activeMenu) {
      setActiveMenu(false);
    } else {
      setActiveMenu(id);
    }
  };

  return (
    <>
      <Wrapper active={active} ariaHidden={active}>
        <MobileMenuClose
          type="button"
          onClick={() => {
            openState();
            if (searchFocus) searchState();
          }}
          active={active}
          aria-expanded={active}
        >
          <MenuSVG src={CloseSVG} />
        </MobileMenuClose>
        <SearchDonate resolution="mobile" searchFocus={searchFocus} />
        <Menus role="navigation" aria-label="Main menu">
          {navigationItems && (
            <MenuList role="menubar" aria-hidden="false">
              {navigationItems.map((item, i) => (
                <NavigationMenu
                  key={i}
                  id={item.id}
                  pageData={item}
                  menuOpen={activeMenu === item.id}
                  setActiveMenu={updateActiveMenu}
                />
              ))}
            </MenuList>
          )}
          {additionalLink && (
            <AdditionalMenu role="navigation" aria-label="Secondary menu">
              <Item>
                <ItemLink internalLink={additionalLink[0]}>
                  {additionalLink[0].menuLabel || additionalLink[0].title}
                </ItemLink>
              </Item>
              <Item>
                <ItemLink externalUrl="https://scotland.shelter.org.uk/">
                  Scotland
                </ItemLink>
              </Item>
            </AdditionalMenu>
          )}
        </Menus>
      </Wrapper>
      <Overlay
        active={active}
        subMenuActive={!!activeMenu}
        onClick={() => updateActiveMenu()}
      />
    </>
  );
};

Navigation.propTypes = {
  active: PropTypes.bool.isRequired,
  searchFocus: PropTypes.bool,
  openState: PropTypes.func,
  searchState: PropTypes.func,
  pageData: PropTypes.shape({
    navigationItems: PropTypes.array,
  }),
};

export default Navigation;
