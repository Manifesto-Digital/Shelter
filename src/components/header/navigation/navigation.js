import React from 'react';
import PropTypes from 'prop-types';

import { SubNavButton, ArrowSVG } from './styles-icons';
import { Menu, SubMenu, ItemLink, Item } from './styles';
import AngleRight from '../../../assets/svg/icons/chevron-down-light.svg';

const NavigationMenu = ({ pageData, id, menuOpen, setActiveMenu }) => {
  const { menuLabel, navigationLink, subNavigationItems } = pageData;

  return (
    <Menu>
      {navigationLink &&
        navigationLink.map((navLink, i) => (
          <Item key={i}>
            <ItemLink internalLink={navLink}>
              {menuLabel || navLink.title}
            </ItemLink>
            {subNavigationItems && (
              <SubNavButton type="button" onClick={() => setActiveMenu(id)}>
                <ArrowSVG src={AngleRight} cacheGetRequests />
              </SubNavButton>
            )}
          </Item>
        ))}
      {subNavigationItems && (
        <SubMenu active={menuOpen}>
          {subNavigationItems.map((item, i) => (
            <ItemLink key={i} internalLink={item}>
              {item.title}
            </ItemLink>
          ))}
        </SubMenu>
      )}
    </Menu>
  );
};

NavigationMenu.propTypes = {
  pageData: PropTypes.shape({
    navigationLink: PropTypes.arrayOf(
      PropTypes.shape({
        title: PropTypes.string.isRequired,
        slug: PropTypes.string.isRequired,
      })
    ),
  }),
  id: PropTypes.string.isRequired,
  menuOpen: PropTypes.bool,
  setActiveMenu: PropTypes.func,
};

export default NavigationMenu;
