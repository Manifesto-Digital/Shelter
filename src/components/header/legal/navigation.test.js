import React from 'react';
import 'jest-styled-components';
import { snapshotComponent, mountWithTheme } from 'test-helpers';
import {
  createHeaderNavigation,
  createInternalRef,
} from '../../../utils/test-factories';
import LegalNavigation from './navigation';
import NavigationMenuItem from '../navigation/menu-item';
import { MobileMenuClose, SubNavButton } from '../navigation/styles-icons';
import { ItemLink, SubMenuUl } from '../navigation/styles';
import { hidePascalCaseWarning } from '../../../utils/test-mocks';

it('Renders correctly', () => {
  const mockData = createHeaderNavigation({ acitve: false });
  snapshotComponent(
    <LegalNavigation navigationItems={mockData.navigationItems} active />
  );
});

hidePascalCaseWarning();

it('Displays close button', () => {
  const mockData = createHeaderNavigation();
  const wrapper = mountWithTheme(
    <LegalNavigation navigationItems={mockData.navigationItems} active />
  );
  expect(wrapper.find(MobileMenuClose)).toHaveLength(1);
});

test('Should populate a menu item correctly', () => {
  const mockMenuItem = {
    id: 'e230d8b8-4ee6-5d4c-bf25-57af664d12d7',
    menuLabel: 'What we do',
    navigationLink: [createInternalRef({ title: 'title' })],
  };

  const wrapper = mountWithTheme(
    <NavigationMenuItem menuItem={mockMenuItem} id={mockMenuItem.id} />
  );

  expect(
    wrapper
      .find(ItemLink)
      .at(0)
      .text()
  ).toBe(mockMenuItem.menuLabel);

  expect(
    wrapper
      .find(ItemLink)
      .at(0)
      .props().internalLink
  ).toBe(mockMenuItem.navigationLink[0]);
});

test('Should populate a menu', () => {
  const mockData = createHeaderNavigation({
    id: 'e230d8b8-4ee6-5d4c-bf25-57af664d12d7',
    navigationItems: [
      {
        id: 'd582da1f-1f7f-5f4c-ae3b-ef2b41972dbc',
        menuLabel: 'Label 1',
        navigationLink: [
          {
            title: 'Test Page ',
            slug: 'Test-page',
          },
        ],
      },
      {
        id: 'd582da1f-1f7f-5f4c-ae3b-ef2b41972dbc',
        menuLabel: 'Label 2',
        navigationLink: [
          {
            title: 'Test Page ',
            slug: 'Test-page',
          },
        ],
      },
    ],
    additionalLink: null,
  });
  const wrapper = mountWithTheme(
    <LegalNavigation
      navigationItems={mockData.navigationItems}
      active={false}
      id={mockData.id}
    />
  );

  expect(wrapper.find(ItemLink)).toHaveLength(
    mockData.navigationItems.length + 1
  ); // Scotland link
});

it('Opens a sub-navigation on button click', () => {
  const mockData = createHeaderNavigation({
    navigationItems: [
      {
        id: 'd582da1f-1f7f-5f4c-ae3b-ef2b41972dbc',
        menuLabel: 'Housing Advice',
        navigationLink: [{ title: 'test Page', slug: 'test-page' }],
        childNavigationItems: [
          {
            slug: '320,000-people-in-britain-are-now-homeless',
            title: '320,000 people in Britain are now homeless',
          },
          {
            title: 'test Page',
            slug: 'test-page',
          },
          {
            title: 'Shelter Manchester to hold vigil for lost homeless people',
            slug: 'shelter-manchester-to-hold-vigil-for-lost-homeless-people',
          },
          {
            title: 'Shelter Demo Page',
            slug: 'shelter-demo-page',
          },
        ],
      },
    ],
  });

  const wrapper = mountWithTheme(
    <LegalNavigation navigationItems={mockData.navigationItems} active />
  );
  const mockButton = wrapper.find(SubNavButton);

  mockButton.simulate('click');
  expect(wrapper.find(SubMenuUl)).toHaveStyleRule('display', 'flex');
});
