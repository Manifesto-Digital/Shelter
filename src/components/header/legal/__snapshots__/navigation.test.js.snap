// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  .c1 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 45px;
  height: 40px;
  padding: 0;
  margin-left: auto;
  cursor: pointer;
}

.c10 {
  padding: 0;
  border: none;
  font: inherit;
  color: inherit;
  background-color: transparent;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 45px;
  height: 40px;
  padding: 0;
  margin-left: auto;
  cursor: pointer;
  height: auto;
  width: 25px;
}

.c10 svg {
  width: 15px;
}

.c2 {
  display: block;
  width: 20px;
}

.c11 {
  display: block;
  fill: #212122;
  width: 20px;
  cursor: pointer;
}

.c0 {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  width: 80%;
  height: 100%;
  background: #fff;
  box-shadow: -1px 0 14px rgba(0,0,0,0.5);
  -webkit-transition: -webkit-transform 0.3s cubic-bezier(0,0,0.3,1);
  -webkit-transition: transform 0.3s cubic-bezier(0,0,0.3,1);
  transition: transform 0.3s cubic-bezier(0,0,0.3,1);
  -webkit-transform: translateX(0);
  -ms-transform: translateX(0);
  transform: translateX(0);
  z-index: 10;
  overflow-y: scroll;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  list-style: none;
  margin: 0;
  padding: 0;
}

.c3 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  height: 100%;
  padding: 0;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0;
  padding: 10px;
  border-bottom: 1px solid #e5e5e5;
}

.c7 .c4 {
  padding: 5px 10px;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0;
  padding: 10px;
  border-bottom: 1px solid #e5e5e5;
}

.c14 .c4 {
  padding: 5px 10px;
}

.c15 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-bottom: 0;
  padding: 10px;
  border-bottom: 1px solid #e5e5e5;
}

.c15 .c4 {
  padding: 5px 10px;
}

.c9 {
  color: #212122;
  font-size: 0.775rem;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c12 {
  display: none;
}

.c13 {
  display: none;
  width: 100%;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding-left: 10px;
  border: none;
  list-style: none;
}

.c13 .c6:last-child {
  border: none;
}

.c18 {
  display: block;
  width: 100%;
  max-width: 1140px;
  height: 0;
  margin: 0 auto;
  overflow: hidden;
}

.c18:focus {
  height: auto;
}

.c17 {
  padding: 0;
  margin: 0;
  -webkit-clip: rect(1px,1px,1px,1px);
  clip: rect(1px,1px,1px,1px);
  -webkit-clip-path: inset(1px);
  clip-path: inset(1px);
  width: 1px;
  height: 1px;
  overflow: hidden;
  position: absolute;
  white-space: nowrap;
}

.c16 {
  display: inline-block;
  text-align: center;
  padding: 10px 1em;
  font-weight: normal;
  background-color: #12b9bf;
  border: none;
  color: #fff;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin-right: 10px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c16:hover {
  color: #fff;
}

.c16:hover {
  color: #fff;
}

@media (min-width:62em) {
  .c1 {
    display: none;
  }
}

@media (min-width:62em) {
  .c10 {
    display: none;
  }
}

@media (min-width:62em) {
  .c10 {
    display: none;
    margin-left: 5px;
  }

  .c10 svg {
    width: auto;
  }
}

@media (min-width:62em) {
  .c0 {
    position: relative;
    width: 100%;
    background: #e5e5e5;
    box-shadow: none;
    -webkit-transition: none;
    transition: none;
    -webkit-transform: translateX(0);
    -ms-transform: translateX(0);
    transform: translateX(0);
    overflow-y: visible;
  }
}

@media (min-width:62em) {
  .c5 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    width: auto;
  }
}

@media (min-width:62em) {
  .c3 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
    max-width: 1140px;
    margin: 0 auto;
    padding: 0 10px;
  }
}

@media (min-width:62em) {
  .c7 .c4 {
    padding: 5px 0;
  }
}

@media (min-width:62em) {
  .c7 .c4 {
    margin-right: auto;
    padding: 0;
  }
}

@media (min-width:62em) {
  .c14 .c4 {
    padding: 5px 0;
  }
}

@media (min-width:62em) {
  .c14 .c4 {
    margin-right: auto;
    padding: 10px;
  }
}

@media (min-width:62em) {
  .c15 {
    padding: 0;
  }
}

@media (min-width:62em) {
  .c15 .c4 {
    padding: 5px 0;
  }
}

@media (min-width:62em) {
  .c15 .c4 {
    margin-right: auto;
    padding: 0;
  }
}

@media (min-width:62em) {
  .c13 {
    position: absolute;
    display: none;
    top: 52px;
    right: 0;
    left: 0;
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
    -webkit-flex-wrap: wrap;
    -ms-flex-wrap: wrap;
    flex-wrap: wrap;
    width: 100%;
    height: auto;
    margin: 0;
    background: #d9d9d9;
    padding: 10px 10px;
  }
}

@media (min-width:62em) {
  .c13 .c6 {
    -webkit-flex: 0 1 calc(33.333333333333336% - 10px);
    -ms-flex: 0 1 calc(33.333333333333336% - 10px);
    flex: 0 1 calc(33.333333333333336% - 10px);
    width: 32%;
    margin-left: 10px;
    border: none;
  }
}

@media (min-width:62em) {
  .c13 .c6:hover {
    background: #212122;
  }

  .c13 .c6:hover .c8 {
    color: #fff;
  }
}

@media (min-width:62em) {
  .c16 {
    height: 100%;
  }
}

<div
    className="c0"
  >
    <button
      aria-expanded={true}
      className="c1"
      onClick={[Function]}
      type="button"
    >
      <span
        className="isvg loading c2"
      />
    </button>
    <nav
      aria-label="Main menu"
      className="c3"
      role="navigation"
    >
      <ul
        aria-hidden="false"
        className="c4 c5"
        role="menubar"
      >
        <li
          aria-haspopup={true}
          className="c6 c7"
          role="menuitem"
        >
          <Link
            className="c8 c9"
            to="/test-page"
          >
            Housing Advice
          </Link>
          <button
            aria-expanded={false}
            className="tracking-subnav-button c10"
            data-tracking="Housing Advice"
            onClick={[Function]}
            type="button"
          >
            <span
              className="isvg loading c11"
            />
          </button>
        </li>
        <li
          className="c12"
        >
          <ul
            aria-expanded={false}
            aria-hidden={true}
            className="c13"
            role="menu"
          >
            <li
              className="c6 c14"
            >
              <Link
                className="c8 c9"
                to="/sprint-7-legal-page-demo"
              >
                Benefits
              </Link>
            </li>
          </ul>
        </li>
        <li
          aria-haspopup={false}
          className="c6 c7"
          role="menuitem"
        >
          <Link
            className="c8 c9"
            to="/test-page"
          >
            What we do
          </Link>
        </li>
        <li
          className="c6 c15"
        >
          <Link
            className="c16"
            to="/donate"
          >
            Donate
            <legend
              className="c17"
            >
              Donate
            </legend>
          </Link>
        </li>
        <li
          className="c6 c15"
        >
          <a
            className="c8 c9"
            href="https://scotland.shelter.org.uk/"
            rel="noopener noreferrer"
            target="_blank"
          >
            Scotland
          </a>
        </li>
      </ul>
    </nav>
    <a
      className="c18"
      href="#main"
    >
      Skip to main content
    </a>
  </div>,
  .c0 {
  display: block;
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 5;
  background: rgba(43,46,56,0.7);
  pointer-events: all;
  opacity: 1;
  -webkit-transition: 0.2s ease-in opacity;
  transition: 0.2s ease-in opacity;
}

@media (min-width:62em) {
  .c0 {
    width: 100%;
    opacity: 0;
    pointer-events: none;
  }
}

<div
    className="c0"
    onClick={[Function]}
  />,
]
`;
